In [1]:
import pandas as pd
import os
import numpy as np
import matplotlib.pyplot as plt
import plotly
plotly.tools.set_credentials_file(username='paranyaj',api_key='RYp3Uubq9Tgbeogsz3jF')



path = '/Users/BOWPJ/Desktop/NVTC'
os.chdir(path)


# Data from: https://catalog.data.gov/dataset/rural-veterans-by-state-e6abb
rural_urban = pd.read_excel('Rural_Veterans_by_State_2014.xlsx')

rural_urban = rural_urban[2:]
rural_urban = rural_urban.reset_index()

rural_urban.columns = ['state','vetpop_rural','vetpop_urban','perc_f_rural','perc_f_urban','unemp_vet_rural','unemp_vet_urban','bel_pov_inc_rural','bel_pov_inc_urban',
                       'disabil_rate_rural','disabil_rate_urban','med_inc_rural','med_inc_urban',
                       '18-34_rural','18-34_urban','35-54_rural','35-54_urban','55-64_rural','55-64_urban',
                       '65-74_rural','65-74_urban','75+_rural','75+_urban','gulf2_rural','gulf2_urban',
                       'gulf1_rural','gulf1_urban','vietnam_rural','vietnam_urban','korean_rural','korean_urban',
                       'world2_rural','world2_urban','less_highschool_rural','less_highschool_urban','highschool_rural',
                       'highschool_urban','some_college_rural','some_college_urban','bachelor_rural','bachelor_urban']
                      

rural_urban = rural_urban.replace('X',np.nan)
rural_urban = rural_urban[:52]


In [2]:
rural_urban.head()

state  vetpop_rural  vetpop_urban  perc_f_rural  perc_f_urban  \
0     Alabama      142884.0      201420.0           7.6           9.0   
1      Alaska       21470.0       48378.0           8.4          15.0   
2     Arizona       62905.0      420121.0           7.0           8.3   
3    Arkansas      101788.0      106098.0           6.3           9.6   
4  California      142266.0     1551336.0           5.4           7.7   

   unemp_vet_rural  unemp_vet_urban  bel_pov_inc_rural  bel_pov_inc_urban  \
0              4.0              6.5                8.5                8.8   
1             12.3              6.4                4.9                4.0   
2              5.3              7.0                8.4                7.4   
3              4.2              7.6                9.1                8.1   
4              8.6              7.8                8.8                7.6   

   disabil_rate_rural       ...        world2_rural  world2_urban  \
0                36.0       ...                 3.4           4.7   
1                30.5       ...                 1.3           2.9   
2                34.4       ...                 4.1           5.8   
3                38.7       ...                 5.2           4.2   
4                32.9       ...                 6.4           6.7   

   less_highschool_rural  less_highschool_urban  highschool_rural  \
0                   10.1                    5.3              35.2   
1                    4.1                    3.8              31.8   
2                    6.2                    5.2              23.2   
3                   11.1                    7.2              38.6   
4                    5.9                    5.8              24.3   

   highschool_urban  some_college_rural  some_college_urban  bachelor_rural  \
0              24.6                37.1                39.0            17.6   
1              19.4                38.0                49.4            26.2   
2              22.1                43.2                42.8            27.4   
3              28.7                36.1                37.2            14.2   
4              19.5                42.5                42.0            27.3   

   bachelor_urban  
0            31.1  
1            27.4  
2            29.9  
3            26.9  
4            32.7  

[5 rows x 41 columns]

# Characteristics (%) of veterans in each state

In [3]:
# 2014 data from: https://www.census.gov/library/visualizations/2015/comm/veterans-statistics.html
characteristics = pd.read_excel('vet-state-characteristics.xlsx')
characteristics = characteristics.dropna(axis=1,thresh=20)
characteristics.columns = ['state','total_vet_pop','female','med_inc','gulf2','gulf1','vietnam','korean','world2','peace','bachelor','enrolled',
                          'uninsured','disabil','VA','18-29','30-44','45-54','55-64','65+','-10k','10k-14.9k','15k-24.9k','25k-34.9k',
                          '35k-49.9k','50k-74.9','75k-99.9k','100k-149.9k','150k-199.9k','200k+']

characteristics = characteristics.iloc[2:55]
characteristics = characteristics.iloc[1:]
characteristics.index = range(0,len(characteristics))
characteristics.head()


state total_vet_pop   female med_inc gulf2 gulf1 vietnam korean  \
0     Alabama        388865   7.6911   57935  13.4  14.6    33.6    8.9   
1      Alaska         71004  12.7106   85518  24.2  17.8    30.8    4.2   
2     Arizona        522382  7.69207   60197  11.5  12.1    33.6     11   
3    Arkansas        237311  7.13915   50669  11.8  12.8    34.5    9.7   
4  California       1893539  6.85257   73722  11.4  10.7    33.9   10.8   

  world2 peace  ...   -10k 10k-14.9k 15k-24.9k 25k-34.9k 35k-49.9k 50k-74.9  \
0    5.7  23.8  ...    3.5       3.6       8.9      10.7      15.8       21   
1    2.5  20.5  ...    1.7       2.4       4.6       5.8      10.1     17.7   
2    7.3  24.5  ...   3.36      3.03       8.2     10.07     15.44    21.44   
3    6.2    25  ...   4.02      4.36     11.12     12.29     17.43     22.2   
4    8.5  24.7  ...   2.61      2.94      6.61      7.91      12.1    18.69   

  75k-99.9k 100k-149.9k 150k-199.9k 200k+  
0      14.3          14         4.7   3.6  
1      16.8        22.8          11   7.1  
2     14.47          15        5.15  3.83  
3     12.55       10.94        2.93  2.16  
4     14.58       18.64         8.3  7.61  

[5 rows x 30 columns]

In [4]:
suicide_rate = pd.read_excel('suicide_rate.xlsx')
suicide_rate.head()

state  total_vet  total_all  18-34_vet  18-34_all  35-54_vet  \
0     Alabama       37.8       18.8       86.1       16.1       45.7   
1      Alaska       54.4       30.0      101.7       41.2       56.7   
2     Arizona       53.6       23.7       76.8       19.7       74.5   
3    Arkansas       48.1       22.0      116.4       20.1       50.9   
4  California       39.1       14.0       47.1       10.0       45.0   

   35-54_all  55-74_vet  55-74_all  75+_vet  75+_all  
0       22.3       29.3       18.0     28.7     16.7  
1       28.4        NaN        NaN      NaN      NaN  
2       24.9       42.4       25.9     48.3     25.9  
3       25.6       40.5       21.6     35.4     15.8  
4       14.3       32.3       16.5     42.3     21.6

In [5]:
import plotly.plotly as py
import plotly.graph_objs as go
import plotly
plotly.tools.set_credentials_file(username='paranyaj',api_key='RYp3Uubq9Tgbeogsz3jF')
gender_suicide = pd.read_excel('gender_vet_suicide.xlsx')

male = go.Scatter(x=gender_suicide['Year'],y=gender_suicide['Male'],mode='lines+markers',name='Male',
                 line=dict(color='blue'))
female = go.Scatter(x=gender_suicide['Year'],y=gender_suicide['Female'],
                    mode='lines+markers',name='Female',line=dict(
    color='rgba(231, 84,128 , .8)'))
total = go.Scatter(x=gender_suicide['Year'],y=gender_suicide['Total'],mode='lines+markers',name='Total',
                  line=dict(color='green'))
data = [male,female,total]
layout = dict(title='Veteran Suicide Rate From 2001 to 2014',
             yaxis=dict(title='Suicide Rate'),
             xaxis = dict(title='Year'))
fig = dict(data=data,layout=layout)
py.iplot(fig,file='gender_suicide')


## Characteristics + crude suicide rate for each state

In [8]:
import plotly.figure_factory as ff
# Adding suicide rate to characteristics data frame
characteristics['suicide_rate'] = suicide_rate['total_vet']

### Correlation between each characteristic and crude suicide rate

In [9]:
cols = characteristics.columns.drop('state')
characteristics[cols] = characteristics[cols].apply(pd.to_numeric, errors='coerce')
characteristics_nostate = characteristics.drop('state',axis=1)
characteristics_nostate.corr()['suicide_rate'].sort_values(ascending=False)

suicide_rate     1.000000
uninsured        0.572600
50k-74.9         0.516602
55-64            0.516001
vietnam          0.460681
VA               0.395031
gulf1            0.363383
disabil          0.357463
30-44            0.332506
35k-49.9k        0.265868
18-29            0.251435
75k-99.9k        0.194078
female           0.153577
gulf2            0.148778
enrolled         0.148345
25k-34.9k        0.144697
45-54            0.055993
15k-24.9k        0.016018
10k-14.9k        0.015760
-10k            -0.057223
100k-149.9k     -0.163146
bachelor        -0.205426
med_inc         -0.208567
total_vet_pop   -0.266475
150k-199.9k     -0.276720
peace           -0.285262
world2          -0.355355
65+             -0.365545
200k+           -0.384025
korean          -0.407222
Name: suicide_rate, dtype: float64

## Plotting Uninsured vs. Suicide Rate

In [10]:
%matplotlib inline
import pylab

uninsured_scatter = go.Scatter(x=characteristics['uninsured'],
          y = characteristics['suicide_rate'],
                               text = characteristics['state'],
          mode = 'markers',
          marker = dict(size=10,color='blue',line=dict(width=2,color='black')))


layout = dict(title='Percentage of Uninsured Veterans vs. Suicide Rate',
             yaxis=dict(title='Crude Suicide Rate'),
             xaxis = dict(title='Percentage of Uninsured Veterans'))
fig = dict(data=[uninsured_scatter],layout=layout)
py.iplot(fig, filename='uninsured')


High five! You successfully sent some data to your account on plotly. View your plot in your browser at https://plot.ly/~paranyaj/0 or inside your plot.ly account where it is named 'uninsured'


Notice: percentage of uninsured veterans is most highly correlated with suicide rate.

### Do states with more percentage of veterans in rural areas have more suicides?

In [11]:
cols = characteristics.columns.drop('state')
characteristics[cols] = characteristics[cols].apply(pd.to_numeric, errors='coerce')

rate_rural_urban = pd.DataFrame({'state':rural_urban['state'],'suicide_rate':suicide_rate['total_vet']})
rate_rural_urban['percent_rural'] = rural_urban['vetpop_rural']/characteristics['total_vet_pop']
rate_rural_urban['percent_urban'] = rural_urban['vetpop_urban']/characteristics['total_vet_pop']
rate_rural_urban[['suicide_rate','percent_rural','percent_urban']].corr()
rate_rural_urban.tail()



state  suicide_rate  percent_rural  percent_urban
47     Washington          39.2       0.190302       0.769562
48  West Virginia          49.0       0.457315       0.425449
49      Wisconsin          36.1       0.313339       0.587390
50        Wyoming          55.0       0.324392       0.568788
51    Puerto Rico          14.8       0.036117       0.811194

### Which states have the highest percentage of veterans in rural areas?

In [12]:
# Reminder than suicide_rate is crude rate (per 100,000 people)
rate_rural_urban.sort_values(by='percent_rural',ascending=False).head(10)

state  suicide_rate  percent_rural  percent_urban
19          Maine          48.3       0.600006       0.298419
45        Vermont          54.9       0.573613       0.328752
48  West Virginia          49.0       0.457315       0.425449
26        Montana          68.6       0.433191       0.461855
24    Mississippi          42.4       0.429210       0.440039
3        Arkansas          48.1       0.428922       0.447084
41   South Dakota          47.9       0.423755       0.499043
34   North Dakota          45.8       0.391839       0.552251
29  New Hampshire          37.2       0.390425       0.492207
17       Kentucky          48.6       0.375279       0.521204

In [13]:
rate_rural_urban.sort_values(by='suicide_rate',ascending=False).head(10)

state  suicide_rate  percent_rural  percent_urban
26        Montana          68.6       0.433191       0.461855
44           Utah          62.4       0.099749       0.814072
31     New Mexico          59.8       0.206216       0.674155
28         Nevada          59.8       0.079672       0.857469
50        Wyoming          55.0       0.324392       0.568788
45        Vermont          54.9       0.573613       0.328752
1          Alaska          54.4       0.302377       0.681342
36       Oklahoma          53.8       0.327586       0.557861
2         Arizona          53.6       0.120420       0.804241
48  West Virginia          49.0       0.457315       0.425449

Notice: Montana has the highest suicide rate and one of the highest percentage of veterans in rural areas.
2014 suicides in Montana:
* Total suicides: 58
* Age 18-34: < 10
* Age 35-54: 21
* Age 55-74: 10-20
* Total Male: 50-60
* Total Female: <10

In [14]:
rate_rural_urban.sort_values(by='percent_urban',ascending=False).head(10)

state  suicide_rate  percent_rural  percent_urban
8   District of Columbia           NaN            NaN       0.919233
11                Hawaii          30.0       0.070952       0.904595
28                Nevada          59.8       0.079672       0.857469
9                Florida          40.4       0.106930       0.824456
4             California          39.1       0.075132       0.819279
44                  Utah          62.4       0.099749       0.814072
51           Puerto Rico          14.8       0.036117       0.811194
2                Arizona          53.6       0.120420       0.804241
39          Rhode Island          25.1       0.105960       0.796406
21         Massachusetts          21.6       0.094454       0.788602

Ranking (highest to lowest) for percentage of veterans living in urban areas.

In [22]:
%matplotlib inline
import pylab

rural_per = go.Scatter(x=rate_rural_urban['percent_rural'],
          y = rate_rural_urban['suicide_rate'],
          mode = 'markers',
                       name = 'Rural',
                       text=rate_rural_urban['state'],
                       
          marker = dict(size=10,color='red',line=dict(width=2,color='black')))
urban_per = go.Scatter(x=rate_rural_urban['percent_urban'],
                      y = rate_rural_urban['suicide_rate'],
                       name = 'Urban',
                       text=rate_rural_urban['state'],
                      mode='markers',
                      marker = dict(size=10,color='yellow',
                                   line=dict(width=2,color='black')))
data_area = [rural_per,urban_per]

layout = dict(title='Percentage of Veterans in Rural and Urban Areas vs. Suicide Rate',
             yaxis=dict(title='Suicide Rate'),
             xaxis = dict(title='Percentage of Veterans'))
fig = dict(data=data_area,layout=layout)
py.iplot(fig, filename='rural_urban')


### Looking at other characteristics of veterans in states (excluding rural/urban population)

In [16]:
other_rural_char = rural_urban.drop(['vetpop_urban','vetpop_rural','state'],axis=1)
other_rural_char['suicide_rate'] = suicide_rate['total_all']
other_rural_char.corr()['suicide_rate'].sort_values(ascending=False).head(10)

suicide_rate          1.000000
some_college_urban    0.573722
some_college_rural    0.446220
disabil_rate_rural    0.403708
55-64_rural           0.394350
18-34_urban           0.378058
gulf1_urban           0.334110
35-54_urban           0.332889
55-64_urban           0.331035
perc_f_rural          0.320213
Name: suicide_rate, dtype: float64

Correlation between suicide rate in each state for veterans and characteristics of veterans in each state.

### Compare percentage of veteran suicide to general population

In [17]:
diff_all = []
suicide_rate1 = suicide_rate.dropna(subset=['total_vet','18-34_vet','35-54_vet','55-74_vet','75+_vet'],axis=0)
for index, row in suicide_rate1.iterrows():
    if row['total_all'] > row['total_vet']:
        print(row['state'])
    diff = row['total_vet']-row['total_all']
    diff_all.append(diff)
np.mean(diff_all)


21.957142857142859

On average (across all states), the suicide rates for veterans is 21.25% greater than the general population.

In [61]:
diff_1 = []
for index, row in suicide_rate1.iterrows():
    diff = row['18-34_vet']-row['18-34_all']
    diff_1.append(diff)

diff_2 = []
for index, row in suicide_rate1.iterrows():
    diff = row['35-54_vet']-row['35-54_all']
    diff_2.append(diff)
    
diff_3 = []
for index, row in suicide_rate1.iterrows():
    diff = row['55-74_vet']-row['55-74_all']
    diff_3.append(diff)

diff_4 = []
for index, row in suicide_rate1.iterrows():
    diff = row['75+_vet']-row['75+_all']
    diff_4.append(diff)
    
print(np.mean(diff_1))
print(np.mean(diff_2))
print(np.mean(diff_3))
print(np.mean(diff_4))

62.4914285714
29.9285714286
12.9942857143
14.5914285714


Numbers above are the average difference in suicide rate for each age group across states.

### Comparing Suicide Methods between General/Veterans

In [35]:
# suicide rates 2014
# All: Female: 1.5, Male: 2.6

labels = ['Firearm','Suffocation','Poisoning','Other']
f_number = [2999,3296,2533,833]
m_number = [18335,3513,8874,2391]

colors = ['#01579B','#B3E5FC','#03A9F4','#B3E5FC']
fig = {'data':[{'values':[18335,3513,8874,2391],
               'labels':['Firearm','Suffocation','Poisoning','Other'],
               'domain':{'x':[0,0.48]},'name':'General Male Suicide Methods',
               'hoverinfo':'label+percent+name',
               'hole':0.4,'type':'pie'},
              {'values':[2999,3296,2533,833],'labels':['Firearm','Suffocation','Poisoning','Other'],'text':'Female',
              'textposition':'inside','domain':{'x':[0.52,1]},'name':'General Female Suicide Method','hoverinfo':'label+percent"name',
              'hole':0.4,'type':'pie'}],
      'layout':{'title':'General Suicide Methods','annotations':[
            {'font':{'size':20},
            'showarrow':False,'text':'Male','x':0.18,'y':0.5},
            {'font':{'size':20},'showarrow':False,'text':'Female','x':0.85,'y':0.5
            }
        ]}}

pie1 = go.Pie(labels=labels,values=m_number,textinfo='value',textfont=dict(size=10),
             marker=dict(colors=colors, 
                           line=dict(color='black', width=2)))
py.iplot(fig,filename='General Suicide Methods')


In [38]:

colors = ['#01579B','#B3E5FC','#03A9F4','#B3E5FC']
fig = {'data':[{'values':[67.9,10.2,17.2,4.7],
               'labels':['Firearm','Suffocation','Poisoning','Other'],
               'domain':{'x':[0,0.48]},'name':'Veteran Male Suicide Methods',
               'hoverinfo':'label+percent+name',
               'hole':0.4,'type':'pie'},
              {'values':[40.5,32.3,20.2,7],'labels':['Firearm','Suffocation','Poisoning','Other'],'text':'Female',
              'textposition':'inside','domain':{'x':[0.52,1]},'name':'Veteran Female Suicide Method','hoverinfo':'label+percent"name',
              'hole':0.4,'type':'pie'}],
      'layout':{'title':'Veteran Suicide Methods','annotations':[
            {'font':{'size':20},
            'showarrow':False,'text':'Male','x':0.18,'y':0.5},
            {'font':{'size':20},'showarrow':False,'text':'Female','x':0.85,'y':0.5
            }
        ]}}


py.iplot(fig,filename='Veteran Suicide Methods')



In [38]:
suicide_methods_all = [0.502,1-0.502] # Firearms, others

suicide_methods_veterans = pd.read_excel('Methods.xlsx')
for index, row in suicide_methods_veterans.iterrows():
    row['% by Firearm'] = pd.to_numeric(str(row['% by Firearm']).replace('%',''),errors='coerce')
    row['% by Suffocation'] = pd.to_numeric(str(row['% by Suffocation']).replace('%',''),errors='coerce')
    row['% by Poisoning'] = pd.to_numeric(str(row['% by Poisoning']).replace('%',''),errors='coerce')
suicide_methods_veterans = suicide_methods_veterans.dropna(subset=['% by Firearm'],axis=0,how='any')
suicide_methods_veterans['% by Firearm'] = pd.to_numeric(suicide_methods_veterans['% by Firearm'],errors='coerce')
suicide_methods_veterans.info()
suicide_methods_veterans.head()


<class 'pandas.core.frame.DataFrame'>
Int64Index: 43 entries, 0 to 50
Data columns (total 7 columns):
State               43 non-null object
% by Firearm        42 non-null float64
% by Suffocation    39 non-null float64
% by Poisoning      32 non-null object
Other               30 non-null object
Total               43 non-null object
% by Firarm All     43 non-null float64
dtypes: float64(3), object(4)
memory usage: 2.7+ KB


State  % by Firearm  % by Suffocation % by Poisoning  Other  Total  \
0     Alabama         0.808             0.108          0.085    NaN  1.001   
2     Arizona         0.730             0.143           0.14    NaN  1.013   
3    Arkansas         0.720             0.140           0.14    NaN      1   
4  California         0.595             0.187          0.142  0.076      1   
5    Colorado         0.669             0.180          0.152    NaN  1.001   

   % by Firarm All  
0            0.676  
2            0.554  
3            0.627  
4            0.378  
5            0.500

In [39]:
diff_mean = np.mean(suicide_methods_veterans['% by Firearm']-suicide_methods_veterans['% by Firarm All'])
print(diff_mean)
suicide_methods_veterans[['% by Firearm','% by Firarm All']].head()                  

0.14711904761904765


% by Firearm  % by Firarm All
0         0.808            0.676
2         0.730            0.554
3         0.720            0.627
4         0.595            0.378
5         0.669            0.500

The average percentage of veterans using firearm for suicide is 14.7% higher than the percentage for the general population.

### Difference in means test for firearm use across states

In [40]:
# H0= mean veteran <= mean general
mean_firearm_vet = np.mean(suicide_methods_veterans['% by Firearm'])
mean_firearm_all = np.mean(suicide_methods_veterans['% by Firarm All'])
n_vet = len(suicide_methods_veterans['% by Firearm'])
n_all = len(suicide_methods_veterans['% by Firarm All'])
var_vet = suicide_methods_veterans['% by Firearm'].var()
var_all = suicide_methods_veterans['% by Firarm All'].var()
diff_firearm = mean_firearm_vet-mean_firearm_all

mean_firearm_vet
mean_firearm_all

# Critical calues = 1.645
z_star = diff_firearm/(var_vet/n_vet+var_all/n_all)**(1/2)
z_star

6.975850676468974

z-score = 6.965 > critical value 1.645 -> reject null hypothesis in favor of alternative. The mean percentage of veterans using firearm as suicide methods across states is significantly greater than that of the general population (5% significance level).

### Analyzing PTSD

In [60]:
ptsd = pd.read_excel('PTSD Data.xlsx',sheet_name=2)
ptsd = ptsd.rename(columns={'State':'state'})
ptsd = ptsd.merge(suicide_rate[['state','total_vet']],on='state')
ptsd.head()

ptsd_plot = go.Scatter(x=ptsd['PTSD rate'],
          y = ptsd['total_vet'],
          mode = 'markers',
                       name = 'Rural',
                       text=ptsd['state'],
                       
          marker = dict(size=10,color='rgba(178,34,34,0.8)',line=dict(width=2,color='black')))
layout = dict(title='Percentage of PTSD vs. Suicide Rate',
             yaxis=dict(title='Suicide Rate'),
             xaxis = dict(title='Percentage of Veterans with PTSD'))
fig = dict(data=[ptsd_plot],layout=layout)
py.iplot(fig, filename='PTSD')

In [45]:
ptsd.corr()['total_vet']

Total veterans                        -0.241760
N with PTSD                           -0.234244
PTSD rate                             -0.078026
Average psych stops for PTSD dx >=1   -0.207707
Weighted average                      -0.217865
Average stops for substance abuse     -0.260942
total_vet                              1.000000
Name: total_vet, dtype: float64

There does not seem to be any significant correlation ebtween PTSD rates and suicide rate.

### Looking at Correlation between Unemployment Rate and Suicide Rate

In [59]:
unemp = pd.read_excel('UnemploymentPercent.xlsx')
unemp = unemp[4:]
unemp = unemp.rename(columns={'State':'state'})
unemp = unemp.merge(suicide_rate[['state','total_vet']],how='inner',on='state')
unemp = unemp.drop('Unnamed: 2',axis=1)
unemp['Unemployed']=pd.to_numeric(unemp['Unemployed'],errors='coerce')
unemp.corr()

Unemployed  total_vet
Unemployed    1.000000  -0.186603
total_vet    -0.186603   1.000000

In [63]:
religion = pd.read_excel('religion.xlsx')
cols = ['Buddhist','Catholic','Eastern Orthodox','Ethiopian Orthodox','Greek Orthodox','Hindu','Jewish','Luteran','Mormon','Muslim','Protestant','Shinto','Sikh']
for col in cols:
    religion[col] = pd.to_numeric(religion[col],errors='coerce')
    religion[col] = religion[col]/religion['vet_pop']

religion['Non-Catholic'] = 1-religion['Catholic']
religion = religion[:-1]
religion.tail()
religion.corr()['total_vet']


Buddhist              0.067409
Catholic             -0.187838
Eastern Orthodox     -0.214812
Ethiopian Orthodox    0.296098
Greek Orthodox       -0.426281
Hindu                 0.046811
Jewish               -0.113925
Luteran               0.220763
Mormon                0.302671
Muslim                0.097471
Protestant            0.025294
Shinto               -0.132578
Sikh                  0.203377
Unknown              -0.293390
total_vet             1.000000
vet_pop              -0.323437
Non-Catholic          0.187838
Name: total_vet, dtype: float64

In [64]:
religion = go.Scatter(x=religion['Catholic'],
          y = religion['total_vet'],
          mode = 'markers',
                       name = 'Rural',
                       text=religion['State'],
                       
          marker = dict(size=10,color='rgba(178,34,34,0.8)',line=dict(width=2,color='black')))
layout = dict(title='Percentage of Catholics vs. Suicide Rate',
             yaxis=dict(title='Suicide Rate'),
             xaxis = dict(title='Percentage of Catholics'))
fig = dict(data=[religion],layout=layout)
py.iplot(fig, filename='religion')